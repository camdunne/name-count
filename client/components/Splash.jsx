import React, { Component } from 'react';
import axios from 'axios';

import Form from '../components/Form';
import Chart from '../components/Chart';

class Splash extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fetched: false,
      data: null,
      title: '',
      value: '',
      error: null,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    axios.get('/api/macbeth')
      .then(({ data }) => {
        const { title, k, v } = data;
        this.setState({
          fetched: true,
          data: {
            labels: k,
            datasets: [{
              backgroundColor: 'rgba(255,99,132,0.2)',
              borderColor: 'rgba(255,99,132,1)',
              borderWidth: 1,
              hoverBackgroundColor: 'rgba(255,99,132,0.4)',
              hoverBorderColor: 'rgba(255,99,132,1)',
              data: v,
            }],
          },
          title,
        });
      })
      .catch((err) => { console.error(err); });
  }
  handleChange(event) {
    this.setState({ value: event.target.value });
  }
  handleSubmit(e) {
    e.preventDefault();
    if (/.*\.xml$/.test(this.state.value)) {
      axios.post('/api/xml', { xmlfile: this.state.value })
        .then(({ data }) => {
          this.setState({
            error: null,
            data: data.data,
            title: data.title,
          });
        })
        .catch((err) => { console.error(err); });
    } else {
      this.setState({
        error: 'Not a valid XML file.',
      });
    }
  }
  render() {
    if (!this.state.fetched) {
      return (
        <div>loading...</div>
      );
    }
    return (
      <div>
        <Form
          value={this.state.value}
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
        />
        <center>{(this.state.error) ? this.state.error : ''}</center>
        <h1>{this.state.title}</h1>
        <Chart data={this.state.data} />
      </div>
    );
  }
}


export default Splash;
