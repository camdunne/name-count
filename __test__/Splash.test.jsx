import React from 'react';
import { mount } from 'enzyme';
import { spy } from 'sinon';
import renderer from 'react-test-renderer';
import Splash from '../client/components/Splash';

describe('<Splash />', () => {
  it('calls componentDidMount', () => {
    spy(Splash.prototype, 'componentDidMount');
    const wrapper = mount(<Splash />);
    expect(Splash.prototype.componentDidMount.calledOnce).toBe(true);
  });
  it('allows us to set props', () => {
    const wrapper = mount(<Splash init="splash" />);
    expect(wrapper.props().init).toBe('splash');
    wrapper.setProps({ init: 'not splash' });
    expect(wrapper.props().init).toBe('not splash');
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Splash />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
