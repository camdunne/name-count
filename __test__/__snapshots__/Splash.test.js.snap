// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> renders correctly 1`] = `
<div>
  loading...
</div>
`;
